### PR TITLE
TE-14.1 : Update gribi_scaling_test.go

### PR DESCRIPTION
**Pull request by avaneesh90**
Please find the below update regarding this pull request
1. Added Type configuration Type: oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE
2. For dutPort2 and atePort2, unit 0 subinterface is configured without vlan(vlan-id 0) and subinterfaces 1-63 are configured with vlan. Since vlan-id value can start with 1 as per yang model definition, changed i to 1 so that subinterface and vlan starts with index 1
3. Index 1-64 are used for programming NH which is used for prefixes in default instance. Hence adjusted indexes of NH used for vrf1,vrf2 and vrf3 prefixes.

### CODE REVIEW

File: feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go [hunk 1]
```diff
@@ -298,14 +298,15 @@ func createVrf(t *testing.T, dut *ondatra.DUTDevice, d *oc.Root, vrfs []string)
...
 			i.Enabled = ygot.Bool(true)
-			i.EnabledAddressFamilies = []oc.E_Types_ADDRESS_FAMILY{
-				oc.Types_ADDRESS_FAMILY_IPV4,
-				oc.Types_ADDRESS_FAMILY_IPV6,
-			}
 			i.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, *deviations.StaticProtocolName)
 			gnmi.Replace(t, dut, gnmi.OC().NetworkInstance(vrf).Config(), i)
 			nip := gnmi.OC().NetworkInstance(vrf)
 			fptest.LogQuery(t, "nonDefaultNI", nip.Config(), gnmi.GetConfig(t, dut, nip.Config()))
+		} else if vrf == *deviations.DefaultNetworkInstance {
```
Comment (mojiiba) [inline feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:305]: @avaneesh90  could you please guard this code block with ExplicitInterfaceInDefaultVRF deviation.

**Comment by robshakir** [comment]
I centralised the removal of the `EnabledAddressFamilies` from the entire codebase along with a test to validate this #876 -- I suggest we try and keep this clearly defined together rather than piecemeal doing it across different tests.

Can I ask why we are changing the NH IDs here?

**Comment by mohanasm** [comment]
> I centralised the removal of the `EnabledAddressFamilies` from the entire codebase along with a test to validate this #876 -- I suggest we try and keep this clearly defined together rather than piecemeal doing it across different tests.
> 
> Can I ask why we are changing the NH IDs here?

Index 1-64 are used for programming NH which is used for prefixes in default instance. Hence adjusted indexes of NH used for vrf1,vrf2 and vrf3 prefixes. 

https://github.com/openconfig/featureprofiles/blob/3c8d81b153d3ede9e3da1f55318bca77c1311962/feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go#L253

File: feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go [hunk 2]
```diff
@@ -340,7 +341,7 @@ func generateSubIntfPair(t *testing.T, dut *ondatra.DUTDevice, dutPort *ondatra.
 	nextHops := []string{}
 	nextHopCount := 63 // nextHopCount specifies number of nextHop IPs needed.
 	for i := 0; i <= nextHopCount; i++ {
-		vlanID := uint16(i)
+		vlanID := uint16(i) + 1
```
Comment (sachendras) [inline feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:344]: Can we please add a comment here on why we are adding "1" to VLANs?

Comment (avaneesh90) [reply feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:344]: As per yang file vlan rang is 1-4094 thats what we are using this.
yang file : https://github.com/openconfig/public/blob/b34db05e8cf2efe69df3762d4bbd80665e1f9e79/release/models/vlan/openconfig-vlan-types.yang#L133

File: feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go [hunk 3]
```diff
@@ -302,6 +302,11 @@ func createVrf(t *testing.T, dut *ondatra.DUTDevice, d *oc.Root, vrfs []string)
 			gnmi.Replace(t, dut, gnmi.OC().NetworkInstance(vrf).Config(), i)
 			nip := gnmi.OC().NetworkInstance(vrf)
 			fptest.LogQuery(t, "nonDefaultNI", nip.Config(), gnmi.GetConfig(t, dut, nip.Config()))
+		} else if vrf == *deviations.DefaultNetworkInstance {
+			i := d.GetOrCreateNetworkInstance(vrf)
+			i.Type = oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE
+			i.Enabled = ygot.Bool(true)
```
Comment (sachendras) [inline feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:308]: With the changes in https://github.com/openconfig/featureprofiles/commit/bac3c1844543afcb133401c77d208d3d3f2b90ee, we should remove this line?

Comment (sachendras) [reply feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:308]: Also update line#300

Comment (avaneesh90) [reply feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:308]: @sachendras  could you please paste the line#300, as per current code Line#300 is i.Enabled = ygot.Bool(true).

File: feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go (hunk 1 above)
Comment (sachendras) [reply feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:304]: @avaneesh90 can you please address the comment?

**Comment by ming2004** [comment]
This PR is in the list for this week's effort, and the auto-merge doesn't work. Can you please rebase it so that I can verify it. Thanks.

**Comment by avaneesh90** [comment]
> This PR is in the list for this week's effort, and the auto-merge doesn't work. Can you please rebase it so that I can verify it. Thanks.

I have rebase can you please check

**Comment by avaneesh90** [comment]
> This PR is in the list for this week's effort, and the auto-merge doesn't work. Can you please rebase it so that I can verify it. Thanks.

I have rebased can you please check

File: feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go (hunk 1 above)
Comment (avaneesh90) [reply feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:304]: > @avaneesh90 could you please guard this code block with ExplicitInterfaceInDefaultVRF deviation as well.

@mojiiba - Assigning interface to default network instance is part of below code after creating VRF https://github.com/avaneesh90/featureprofiles/blob/TE-14.1_gRIBI_Scaling/feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go#L437
Please let us know if you still want to guard this block

Comment (sachendras) [reply feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:304]: may be we still require the deviation @mojiiba mentioned.

Comment (mojiiba) [reply feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:304]: please add the guard, we we still need that. thanks a lot!

Comment (ming2004) [reply feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:304]: I do see ExplicitInterfaceInDefaultVRF deviation is used when configuring the default VRF, so the suggestion from @mojiiba should be taken.
However, I don't see Cisco device use this deviation. @mojiiba, can you confirm if this deviation will help Cisco device?

File: feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go [hunk 4]
```diff
@@ -301,6 +301,10 @@ func createVrf(t *testing.T, dut *ondatra.DUTDevice, d *oc.Root, vrfs []string)
 			gnmi.Replace(t, dut, gnmi.OC().NetworkInstance(vrf).Config(), i)
 			nip := gnmi.OC().NetworkInstance(vrf)
 			fptest.LogQuery(t, "nonDefaultNI", nip.Config(), gnmi.GetConfig(t, dut, nip.Config()))
+		} else if vrf == *deviations.DefaultNetworkInstance {
+			i := d.GetOrCreateNetworkInstance(vrf)
+			i.Type = oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE
+			gnmi.Replace(t, dut, gnmi.OC().NetworkInstance(vrf).Config(), i)
```
Comment (ming2004) [inline feature/gribi/ate_tests/gribi_scaling/gribi_scaling_test.go:307]: Creation of the default VRF is not necessary and should be controlled by ExplicitInterfaceInDefaultVRF for other vendor. I can find similar configuration example from other test case. Please modify accordingly.

**Comment by sachendras** [comment]
Replaced by Pull/1067 and Pull/1068

**Pull request closed without merging**